Python/compile.c

```diff
@@ -2280,13 +2280,17 @@ check_compare(struct compiler *c, expr_ty e)
     n = asdl_seq_LEN(e->v.Compare.ops);
     for (i = 0; i < n; i++) {
         cmpop_ty op = (cmpop_ty)asdl_seq_GET(e->v.Compare.ops, i);
-        bool right = check_is_arg((expr_ty)asdl_seq_GET(e->v.Compare.comparators, i));
+        expr_ty right_expr = (expr_ty)asdl_seq_GET(e->v.Compare.comparators, i);
+        bool right = check_is_arg(right_expr);
         if (op == Is || op == IsNot) {
             if (!right || !left) {
                 const char *msg = (op == Is)
-                        ? "\"is\" with a literal. Did you mean \"==\"?"
-                        : "\"is not\" with a literal. Did you mean \"!=\"?";
-                return compiler_warn(c, LOC(e), msg);
+                        ? "\"is\" with '%.200s' literal. Did you mean \"==\"?"
+                        : "\"is not\" with '%.200s' literal. Did you mean \"!=\"?";
+                expr_ty literal = !left ? e->v.Compare.left : right_expr;
```

Comment:
``left`` gets updated in each iteration, but ``e->v.Compare.left`` is always the same thing. Is this correct?

Should we just check the first item before the loop and then here just look at ``right_expr``?

Reply:
I might be missing something, but `bool left = check_is_arg(e->v.Compare.left);` happens once outside the loop and it's only `right` that is updated in each iteration.

Reply:
line 2296?

Reply:
Ah, of course! Thank you so much, this is buggy. I added the fix and a test case to catch the issue.

I think we can't do the first item check before the loop since all comparators may not be `is`, e.g. `x == 3 is y`, but let me know if I'm missing something!